### DISCUSSION TITLE
RX-V475 and scenes. Alternative: Remote?

### DISCUSSION THREAD

**Discussion by ChavezD** [mvdwetering/yamaha_ynca#287] [General]
Hi and first of all, thanks for this great integration!

Tonight I started expanding my HiFi-Automations and wanted to boot my TV via the AVR, as the TV doesn't support wake-on-lan or anything likewise. The only option to not use it's remote control seems to be one of the "SCENE" buttons.
 
Unfortunately it's already written in the documentation, that this doesn't work with my RX-V475. Instead I found the description on "remote control", but I don't have any entity to choose from. Does it have to be enabled manually?

What I have tried so far to power up the TV:
- Media player 'Select source' on RX-V475 MAIN
- Call a service 'Switch: Turn on' on RX-V475 MAIN HDMI Out 1
- Button 'Press' on RX-V475 MAIN TV
(- Remote: Send command) --> doesn't find the receiver.

Any hint or help is highly appreciated! :)

**Comment by mvdwetering** [comment]
The remote entities are disabled by default. You can enable them in the integration page for the "Zone" device.

In the Control section there are 1 or more disabled entities. There is a small link with "+2 entities not shown" (or something similar), click on it and it wil show the disabled entities. Now you can click on the entity and then the cogwheel and you can toggle the swtich to enable.
![image](https://github.com/mvdwetering/yamaha_ynca/assets/732514/3e583fc1-47c9-4c69-946d-289c3c8476f0)
(sorry for Dutch screenshots, but my English dev system is not running right now)

Hmm, actually maybe clicking "28 entities" on the main integration page is easier, then just search for remote and enable it from there.
![image](https://github.com/mvdwetering/yamaha_ynca/assets/732514/1f8855e9-ba43-4e11-8db7-6869d492fc2f)

![image](https://github.com/mvdwetering/yamaha_ynca/assets/732514/880a49c0-d245-48a0-a7e3-2c56b6443e09)

**Reply by ChavezD** [reply]
Hi neighbour, don't worry, some basic word's I'm able to understand.

Wow that's literally the best improvement to my automations ever! 
Now I'm able to boot up the TV in advance - as it's quite slow and always needs to update it's Android TV stuff first. Sometimes even leading Netflix to close while watching a movie. (It is an couple of years old Xiaomi that hasn't seen Firmware updates since day 1). 
And I prefer to save the 20€ in electricity per year that it's consuming in standby - by simply switching all electric plugs off when I leave home. 

Also it's really convinient now to change scenes including all set up sound modes etc.

Best regards from the Nederrijn and have a nice and sunny weekend!

**Comment by ChavezD** [comment]
Btw from the Q&A I can confirm it is working:

> It might be possible to send scene commands using the remote entity as a workaround. Please drop a message in the [discussions](https://github.com/mvdwetering/yamaha_ynca/discussions) if this actually works as I don't have an RX-V475 to test with.

**Reply by mvdwetering** [reply]
Cool, thanks for the confirmation on that. I will update the readme.

**Comment by janveeke** [comment]
Hi! follow up on this one:  I have an Yamaha RX-A6A and wants to use your YNCA integration to change Scenes, as I understand from the documentation should be possible. In general the integration works fine, as I can set a specific volume for instant…. Now I tried two  scripts to try this out: 


alias: Change Yamaha Scene
sequence:
  - data:
      scene: Radio
    target:
      entity_id: media_player.rx_a6a_main
    action: yamaha.select_scene

This one gives an error: “Action yamaha.select_scene uses action script.change_yamaha_scene which was not found.”


I also tried the following:

script:
  change_yamaha_scene:
    alias: "Change Yamaha Scene"
    sequence:
      - service: yamaha_ynca.send_raw_ynca
        data:
          command: "SCN"
          value: "Radio" 

This one gives no error, but the receiver does not respond to it 

Any idea if this should work and what I do wrong? I can see that the "RX-A6A MAIN Remote" entity is enabled, as you suggested above.

**Reply by janveeke** [reply]
I have been busy with this for days, but got it working just after i submitted the above. So leave it, it works now. Just use the original names "scene_1" till 4, and not the scenes 5-8 but i can live with that. Tx for the great service. I need the scenes as it it the only way for to change the "Speaker Setup" configurations, and there is no direct command for that as far as i know. But this will do . 
A happy man

**Reply by mvdwetering** [reply]
I am wondering if you tried configuring the amount of scenes in the Integration to get the scene button entities?
The amount scenes can not be autodetected anymore on newer models, but you can configure the amount of scenes manually.

I think scene buttons work with the A6A. I only ever heard the scene buttons not working on the RX-V475, but maybe people are not using them or did not figure out how to configure.

You can try it by using the "Configure" button on the integration
![image](https://github.com/user-attachments/assets/4b1801aa-9378-4250-94f7-05f1446caf17)

![image](https://github.com/user-attachments/assets/357575ce-f167-4d98-941b-31275b5021f5)
Select "Next" to go to the settings for the Main zone


![image](https://github.com/user-attachments/assets/a20563a5-348e-411f-9c58-29df4e0a1c81)
Then change the amount of scenes from "Autodetect" to the amount supported by your receiver (which sounds like it is 8 from your comment)

Then keep pressing "Next" for the settings of the other zones and select "Submit" for the last one.
Now the integration will reload and create buttons for each scene. 

Pressing the scene button entities will/should activate the scene like it would by pressing the button on the receiver. 
Or you can use the "button.press" action to activate it from an automation or script

If this does not work I could have a look to see if there are remote codes for scenes bigger then 4

**Reply by mvdwetering** [reply]
btw what do you mean with

> "Speaker Setup" configurations

Is that like changing the speaker configuration between 5.1 setup and 7.1 (or one of the other configurations)?

**Reply by janveeke** [reply]
Ah, even better, Had not seen that one, but configured it to 8 now. Excellent. Regarding the speaker setup: it might be an RX-A feature only. As i have 9 channels, it is convienient that i can configure them differently and store that setup as "pattern". So i have a setup in my livingroom with all 9 speakers in full atmos, and an other pattern where i use two of those channels for the kitchen speakers. Just by switching between those two patterns i have sound in my kitchen or not. The weird thing is that you cannot switch it on the remote, but you can make it part of a scene. See below: 
![image](https://github.com/user-attachments/assets/34d845ff-e178-42b3-bb72-3e9f7dcc4490)

![image](https://github.com/user-attachments/assets/35d7ed2c-964f-418d-a2c0-ea05df92f79b)

**Reply by mvdwetering** [reply]
That is an interesting feature.
I have seen a command that probably allows to switch speaker patterns. My receiver does not support it, but if you want we can have a look to see if I can add a dropdown to the integration where you can select speaker Pattern1 or Pattern2.

For that I would need some info from the commands that the receiver is sending.

To get the required data I would need you to:
* Change the pattern
* Change the pattern back to what it was
* Send the following commands (one at the time) with the "Yamaha (YNCA):  Send raw YNCA command" action from the Home Assistant developer tools. The first one should read the current pattern from the receiver, the second bullet probably activates a Pattern. Choose the pattern that is not currently active
  *  `@SYS:SPPATTERN=?`
  *  `@SYS:SPPATTERN=Pattern 1` or `@SYS:SPPATTERN=Pattern 2` select the one that is not active

![image](https://github.com/user-attachments/assets/a78dbe96-24cf-46c8-ad50-d0a074586253)



After doing all that, download the diagnostics file from Home Assistant and attach the file here. That diagnostics file will have a log of the commands sent and their responses so I can verify if it works as expected.

![image](https://github.com/user-attachments/assets/91558112-013f-4e60-a3b5-33300911a8ec)

If it all works fine then I can add it to the integration.

**Reply by janveeke** [reply]
that would be really awesome. It is weird that no one else is talking about this feature, as indeed it is awesome when you have so many channels. (I once only read a review of someone who said that it is that great that it is a pitty only to have two patterns instead of 4 or 8; lot of settings could be stored for different situations). Anyway, I did what you requested, and as far as i can see in the bottom of the log file i indeed see the switching with my remote (on the Yamaha UI on screen) and I see the command i send. The receiver did not switch from pattern 1 to 2 however, after my raw command. Let me know if I have to do any other test; happy to assist to this amazing integration
[RX-A6A.log](https://github.com/user-attachments/files/18437553/RX-A6A.log)

**Reply by janveeke** [reply]
And oh yes, as i am a relative new-bee, I made the beginners fault to ignore the case sensitivity. When i use: raw_data: "@SYS:SPPATTERN=Pattern 2"  it works like a charm. Unbelievable.

**Reply by mvdwetering** [reply]
Can you try sending the command `@SYS:SPPATTERN=?` again?
From the logs it looks like you missed a T in the word pattern when you sent it so the receiver did not recognize it.

The rest of the commands seem fine. Just want to make sure the receiver responds properly to that command also.

**Reply by janveeke** [reply]
I must have had a bad sleep that night. You are right. Here the snippet from the log you are looking for. 

        "967351.716192 Send: @SYS:MODELNAME=?",
        "967351.718631 Received: @SYS:MODELNAME=RX-A6A",
        "967375.752849 Send: @SYS:SPPATTERN=?",
        "967375.754994 Received: @SYS:SPPATTERN=Pattern 2",
        "967405.853557 Send: @SYS:MODELNAME=?",
        "967405.855659 Received: @SYS:MODELNAME=RX-A6A",

**Reply by mvdwetering** [reply]
Thanks for the update. Everything looks like it should. 
I can now add support for it and be confident it will work.

Should be ready soon.

**Reply by mvdwetering** [reply]
I just released v7.14.0 which adds support for selecting "Speaker Pattern".

You will get an additional select entity. Looks like this:
![image](https://github.com/user-attachments/assets/4e930e2b-a648-4ef2-8f70-66f9312859f0)

**Reply by janveeke** [reply]
Really excellent; now I don't need the scenes at all. Tx a ton
